pkg/rehearse/jobs_test.go

```diff
@@ -325,67 +462,58 @@ func TestExecuteJobs(t *testing.T) {
 					[]string{"arg1", fmt.Sprintf("--git-ref=%s@master", anotherTargetRepo)},
 				),
 			},
+		}, {
+			description:  "no jobs",
+			jobs:         map[string][]prowconfig.Presubmit{},
+			expectedJobs: []pjapi.ProwJob{},
+		}, {
+			description: "no rehearsable jobs",
+			jobs: map[string][]prowconfig.Presubmit{
+				targetRepo: {*makeTestingPresubmit("job1", "ci/prow/job1", []string{"--git-ref"})},
+			},
+			expectedJobs: []pjapi.ProwJob{},
 		},
 	}
 
 	for _, tc := range testCases {
 		t.Run(tc.description, func(t *testing.T) {
+			testLogger := logrus.New()
 			fakecs := fake.NewSimpleClientset()
 			fakeclient := fakecs.ProwV1().ProwJobs(testNamespace)
 			watcher, err := fakeclient.Watch(metav1.ListOptions{})
 			if err != nil {
 				t.Fatalf("Failed to setup watch: %v", err)
 			}
-			fakecs.Fake.PrependWatchReactor("prowjobs", func(clientgo_testing.Action) (bool, watch.Interface, error) {
-				watcher.Stop()
-				n := 0
-				for _, jobs := range tc.jobs {
-					n += len(jobs)
-				}
-				ret := watch.NewFakeWithChanSize(n, true)
-				for event := range watcher.ResultChan() {
-					pj := event.Object.(*pjapi.ProwJob).DeepCopy()
-					pj.Status.State = pjapi.SuccessState
-					ret.Modify(pj)
-				}
-				return true, ret, nil
-			})
-			success, err := ExecuteJobs(tc.jobs, testPrNumber, testRepo, testRefs, true, testLogger, fakeclient)
+			fakecs.Fake.PrependWatchReactor("prowjobs", makeSuccessfulFinishReactor(watcher, tc.jobs))
+			success, err := ExecuteJobs(tc.jobs, testPrNumber, testRepoPath, testRefs, true, testLogger, fakeclient)
 
-			if tc.expectedError && err == nil {
-				t.Errorf("Expected ExecuteJobs() to return error")
+			if err != nil {
+				t.Errorf("Expected ExecuteJobs() to not return error, returned %v", err)
 				return
 			}
 
-			if !tc.expectedError {
-				if err != nil {
-					t.Errorf("Expected ExecuteJobs() to not return error, returned %v", err)
-					return
-				}
-
-				if tc.expectedSuccess != success {
-					t.Errorf("Expected ExecuteJobs() to return success=%t, got %t", tc.expectedSuccess, success)
-				}
+			if !success {
+				t.Errorf("Expected ExecuteJobs() to return success=true, got false")
+			}
 
-				createdJobs, err := fakeclient.List(metav1.ListOptions{})
-				if err != nil {
-					t.Errorf("Failed to get expected ProwJobs from fake client")
-					return
-				}
+			createdJobs, err := fakeclient.List(metav1.ListOptions{})
+			if err != nil {
+				t.Errorf("Failed to get expected ProwJobs from fake client")
+				return
+			}
 
-				// Overwrite dynamic struct members to allow comparison
-				for i := range createdJobs.Items {
-					createdJobs.Items[i].Name = generatedName
-					createdJobs.Items[i].Status.StartTime.Reset()
-				}
+			// Overwrite dynamic struct members to allow comparison
+			for i := range createdJobs.Items {
```

Comment:
You could also just validate `[]prowapi.ProwJobSpec`

Reply:
Yeah, that simplifies everything a bit! :+1: